ingestion/src/metadata/ingestion/api/step.py

```diff
@@ -211,7 +211,7 @@ def run(self) -> Iterable[Optional[Entity]]:
         in order to create a generator with `yield`.
         """
         try:
-            for result in self._iter():
+            for result in self._iter() or []:
```

Comment:
not sure if we want this. Might be better to let the source blow up if we don't implement properly returning a generator

Reply:
Makes sense I'll remove that